Plugins~/Src/MeshSyncClientBlender/python/unity_mesh_sync_installation.py

```diff
@@ -16,19 +19,73 @@
 msb_context = ms.Context()
 
 unity_process = None
+meshsync_version = None
 
 def MS_MessageBox(message = "", title = "MeshSync", icon = 'INFO'):
     def draw(self, context):
         self.layout.label(text=message)
     bpy.context.window_manager.popup_menu(draw, title = title, icon = icon)
 
+def msb_version_match_major_minor(v1, v2):
+    if v1 is None:
+        return v2
+    if v2 is None:
+        return v1
+
+    v1.replace("-preview", "")
+    v2.replace("-preview", "")
+    tokens1 = v1.split('.')
+    tokens2 = v2.split('.')
+    return float(tokens1[0]) == float(tokens2[0]) and float(tokens1[1]) == float(tokens2[1])
+
+def msb_find_latest_compatible_version(versions):
+    version_re = re.compile('[0-9]*\.[0-9]*\.[0-9]*-preview')
+
+    latest_version = None
+    for version in versions:
+        if version_re.match(version) is None:
+            continue
+        if msb_version_match_major_minor(version, msb_context.PLUGIN_VERSION):
+            latest_version = msb_get_most_recent_version(latest_version, version)
+    return latest_version
+
+
 def msb_get_meshsync_entry():
-    return "0.15.0-preview"
+
+    global meshsync_version
+    if meshsync_version is not None:
+        return meshsync_version
+
+    ssl._create_default_https_context = ssl._create_unverified_context
+    
+    page = 1
+    while True:
```

Comment:
Why is this in a loop? What are the chances of this becoming an infinite loop?

Reply:
because we have to get results in pages. We exit the loop when the response payload is empty. We could set a max, perhaps to 100 iterations to be safe

Reply:
Done